src/AES.cpp

```diff
@@ -24,29 +24,25 @@ unsigned char *AES::EncryptECB(const unsigned char in[], unsigned int inLen,
                                const unsigned char key[]) {
   CheckLength(inLen);
   unsigned char *out = new unsigned char[inLen];
-  unsigned char *roundKeys = new unsigned char[4 * Nb * (Nr + 1)];
+  unsigned char roundKeys[240];
```

Comment:
Are you fine with using a `static constexpr` private member? or a global variable? (this would be visible in the header file).

```c++
static constexpr unsigned int Nb = 4;
static constexpr unsigned int blockBytesLen = 4 * Nb * sizeof(unsigned char);
```

This way we can still put the `block` and `tmp` array on stack without any C/C++ 11^ standard issues.

```c++
unsigned char block[blockBytesLen];
// and
unsigned char tmp[Nb];
```

If not I'll leave it on heap then.

On the other hand I think I'll leave the `roundKeys` on heap since if we want to move this on stack we might need to change the way on how we initialize the AES class, and use `templates` and `constexpr` expressions that might overly complicate the implementation.

Reply:
Yes, using smth like `static constexpr` in private members will be fine